src/linux/linux-host-session.vala

```diff
@@ -621,19 +650,43 @@ namespace Frida {
 			return process_name.get_string ();
 		}
 
-		public async void start_activity (string package_name, string? activity_name) throws Error {
+		public async void start_package (string package_name, string method, string? entry_point, string? action) throws Error {
+
 			var package_name_value = new Json.Node.alloc ().init_string (package_name);
 
-			var activity_name_value = new Json.Node.alloc ();
-			if (activity_name != null)
-				activity_name_value.init_string (activity_name);
-			else
-				activity_name_value.init_null ();
+			if (method == "activity") {

```

Comment:
Should be using an `enum` here instead of strings.

Reply:
Ok, I suggest EntryPoints for the enum type : 

```
enum EntryPoints {
    ACTIVITY,
    RECEIVER
}
```
Any better idea ?

Reply:
or EntryType ?

Reply:
Perhaps `PackageEntrypoint`? I'd make it `private` and put it right below the method, as it's only used there.